common/lib/xmodule/xmodule/capa_base.py

```diff
@@ -606,28 +606,31 @@ def get_demand_hint(self, hint_index):
         event_info['hint_text'] = get_inner_html_from_xpath(demand_hints[hint_index])
         self.runtime.publish(self, 'edx.problem.hint.demandhint_displayed', event_info)
 
+        # hint_index is the index of the last hint that will be displayed in this rendering,
+        # so add 1 to check if others exist.
+        demand_hint_possible = len(demand_hints) > 0
```

Comment:
If these variables are necessary in 2 methods (here and 659ish), please create a helper method that returns the values.